### PR TITLE
molior-deploy: support EFI ISO installers

### PR DESCRIPTION
**Pull request by neolynx**

### CODE REVIEW

File: plugins/installer.plugin [hunk 1]
```diff
@@ -28,18 +28,41 @@ finalize_deployment_installer()
...
     KERNEL_ARCH="686"
   fi
+
+  if [ "$INSTALLER_BOOTTYPE" = "efi" ]; then
+    INSTALLER_EXTRA_PACKAGES="$INSTALLER_EXTRA_PACKAGES grub-efi-$ARCH-bin dosfstools"
+    if [ "$ARCH" = "amd64" ]; then
+      EFIFILE=bootx64.efi
+      GRUBEFITYPE=x86_64-efi
+    elif [ "$ARCH" = "arm64" ]; then
+      GRUBEFITYPE=arm64-efi
```
Comment (szakalboss) [inline plugins/installer.plugin]: reorder to match above

File: plugins/installer.plugin [hunk 2]
```diff
@@ -53,22 +76,104 @@ finalize_deployment_installer()
...
+    chroot $WORK_DIR/instroot grub-mkimage -O $GRUBEFITYPE -p /boot/grub -o /boot/$EFIFILE \
+                                           part_msdos part_gpt fat iso9660 search
+    if [ $? -ne 0 ]; then
+      log_error "Error creating grub image"
+    fi
+    mkdir -p $WORK_DIR/iso/efi/boot/
+    cp $WORK_DIR/instroot/boot/$EFIFILE $WORK_DIR/iso/efi/boot/
+
+    if [ "$ARCH" = "arm64" ]; then
+      mkdir $WORK_DIR/di
```
Comment (szakalboss) [inline plugins/installer.plugin]: di ?

File: plugins/installer.plugin [hunk 3]
```diff
@@ -28,18 +28,41 @@ finalize_deployment_installer()
...
 
   KERNEL_ARCH="$ARCH"
-
   if [ "$ARCH" = "i386" ]; then
     # The kernel architecture of i386 (32 bit) kernels is called 686
     KERNEL_ARCH="686"
   fi
+
+  if [ "$INSTALLER_BOOTTYPE" = "efi" ]; then
+    INSTALLER_EXTRA_PACKAGES="$INSTALLER_EXTRA_PACKAGES grub-efi-$ARCH-bin dosfstools"
```
Comment (randombenj) [inline plugins/installer.plugin:37]: Why not install the grub-efi meta package instead of arch specific one?

File: plugins/installer.plugin [hunk 4]
```diff
@@ -28,18 +28,41 @@ finalize_deployment_installer()
...
+  fi
+
   if [ "$fs_has_lvm" -eq 1 ]; then
     INSTALLER_EXTRA_PACKAGES="$INSTALLER_EXTRA_PACKAGES lvm2"
   fi
   if [ "$fs_has_luks" -eq 1 ]; then
     INSTALLER_EXTRA_PACKAGES="$INSTALLER_EXTRA_PACKAGES cryptsetup"
   fi
-  chroot $WORK_DIR/instroot apt-get install --no-install-recommends --yes $INSTALLER_EXTRA_PACKAGES linux-image-$KERNEL_ARCH busybox initramfs-tools isolinux syslinux-common xz-utils pv util-linux >&2
+  chroot $WORK_DIR/instroot apt-get install --no-install-recommends --yes \
```
Comment (randombenj) [inline plugins/installer.plugin:55]: image

File: plugins/installer.plugin [hunk 5]
```diff
@@ -53,22 +76,104 @@ finalize_deployment_installer()
   cp /usr/lib/molior-deploy/molior-deploy.sh.inc $initrd/scripts/
 
   if [ "$fs_has_luks" -eq 1 ]; then
+    # FIXME: use manual_add_modules in hooks file ?
     echo "copy_exec /sbin/cryptsetup /sbin" >> $initrd/hooks/installer
     echo dm_crypt >> $initrd/modules
   fi
 
+  if [ "$INSTALLER_BOOTTYPE" = "efi" ]; then
```
Comment (randombenj) [inline plugins/installer.plugin:84]: Please add this as a file in the `initrd-installer` directory.

File: plugins/installer.plugin (hunk 2 above)
Comment (neolynx) [reply plugins/installer.plugin]: debian-installer

**Review by szakalboss — APPROVED** [review]

File: plugins/installer.plugin (hunk 3 above)
Comment (neolynx) [reply plugins/installer.plugin:37]: it is not needed for creating the initrd ;)

File: plugins/installer.plugin [hunk 6]
```diff
@@ -80,20 +185,53 @@ $INSTALLED_PACKAGES
...
   echo " * creating image.cpio.xz"
-  chroot $target sh -c "find / -xdev -depth -print | cpio --quiet -o -H newc -B" > $WORK_DIR/iso/image.cpio
-  $PXZ_CMD -T `nproc --ignore 1` -6 $WORK_DIR/iso/image.cpio 2>&1
+  cd $target
+  find . -xdev -depth -print | cpio --quiet -o -H newc -B > $WORK_DIR/iso/image.cpio
+  if [ $? -ne 0 ]; then
+    log_error "Error creating $WORK_DIR/iso/image.cpio"
+  fi
+  cd - >/dev/null
+  $PXZ_CMD -T `nproc --ignore 1` $WORK_DIR/iso/image.cpio 2>&1
```
Comment (randombenj) [inline plugins/installer.plugin:187]: https://github.com/koalaman/shellcheck/wiki/SC2006

File: plugins/installer.plugin (hunk 2 above)
Comment (randombenj) [reply plugins/installer.plugin]: Rename it to `debian-installer` please :)

File: plugins/installer.plugin (hunk 1 above)
Comment (neolynx) [reply plugins/installer.plugin]: done

File: plugins/installer.plugin (hunk 2 above)
Comment (neolynx) [reply plugins/installer.plugin]: done

File: plugins/installer.plugin (hunk 6 above)
Comment (neolynx) [reply plugins/installer.plugin:187]: agree, but this should go into a separate PR, changing this everywhere. for now it is consistent everywhere, and the issues mentioned there are not problematic here for some reason...

File: plugins/installer.plugin (hunk 5 above)
Comment (neolynx) [reply plugins/installer.plugin:84]: not in this PR... there is more cleanup and improvements needed...

**Review by randombenj — APPROVED** [review]

**Review by hus0 — APPROVED** [review]

**Pull request merged**